### PR TITLE
ISSUE-367 # Multi step support for step file

### PR DESCRIPTION
**Pull request by anishjha93**
# Multi step support for step file

PR Branch
https://github.com/anishjha93/zerocode/tree/anish_step_files

## Checklist:

* [ ✓] Integration tests added?

* [ ✓] Test names are meaningful?

* [ ✓] Feature manually tested?

* [✓ ] Branch build passed?

* [✓ ] No 'package.*' in the imports

### CODE REVIEW

File: core/src/main/java/org/jsmart/zerocode/core/domain/Step.java [hunk 1]
```diff
@@ -26,6 +27,9 @@
     private final JsonNode verify;
     private String id;
     private JsonNode stepFile;
+
+    @JsonFormat(with = JsonFormat.Feature.ACCEPT_SINGLE_VALUE_AS_ARRAY)
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/domain/Step.java]: 👍

File: core/src/test/resources/integration_test_files/multi_step/multi_step_file.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+[
+  {
+    "name": "fist_of_multi_step",
+    "url": "",
+    "operation": "",
+    "customLog": "Contributing to Zerocode",
+    "request": {
+      "req": "12345",
+      "id": "${$.first_test.request.id}"
```
Comment (authorjapps) [inline core/src/test/resources/integration_test_files/multi_step/multi_step_file.json]: 👍

File: core/src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java [hunk 3]
```diff
@@ -162,17 +162,19 @@ private boolean executeSteps(RunNotifier notifier,
...
+            for(Step step : thisSteps) {
+                 wasExecSuccess = executeRetry(notifier,
+                        description,
+                        scenarioExecutionState,
+                        scenario,
+                        step);
+                if (wasExecSuccess != null) {
+                    return wasExecSuccess;
+                }
+            };
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java]: Can you extract this to a method? e.g.
> Boolean executeRetryWithSteps(...)

File: core/src/main/java/org/jsmart/zerocode/core/domain/Step.java (hunk 1 above)
Comment (anishjha93) [reply core/src/main/java/org/jsmart/zerocode/core/domain/Step.java]: Done

File: core/src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java (hunk 3 above)
Comment (anishjha93) [reply core/src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java]: Done

File: core/src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java [hunk 4]
```diff
@@ -161,21 +161,33 @@ private boolean executeSteps(RunNotifier notifier,
...
-                    thisStep);
-
+                    step);
             if (wasExecSuccess != null) {
                 return wasExecSuccess;
             }
         }
-
-        return false;
+        ;
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java]: Can you remote this `;` which seems like a typo.
This shd be the final one I promise, then we will merge it!

**Comment by anishjha93** [comment]
Done

File: core/src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java (hunk 4 above)
Comment (authorjapps) [reply core/src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java]: Can u fix this @anishjha93 ?
> You replied with 'done', but it is still there.

**Pull request merged**